modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/TableViewTableHeaderRowTest.java

```diff
@@ -225,6 +228,70 @@ void testOverriddenColumnPopupMenu() {
         assertNull(columnPopupMenu);
     }
 
+    /**
+     * Tests that re-setting the same columns does not cause memory leaks.
+     * See also: <a href="https://bugs.openjdk.org/browse/JDK-8341687">JDK-8341687</a>.
```

Comment:
technically, this test does not test for memory leak - it seems to test the fact that the listener gets disconnected.